deepspeed/module_inject/replace_module.py

```diff
@@ -373,6 +373,25 @@ def _replace(child, name, conv_linear_layer):
                 return
             mp_replace = ReplaceWithTensorSlicing(mp_group=mp_group)
             weight_shape = child.weight.shape
+            if name == 'attn.Wqkv' and module._get_name() == 'MPTBlock':
+                # MPT block qkv weight's allocation is different from other models, it's [3,num_head,head_dim,hidden_size]
+                # instead of [num_head,3,head_dim,hidden_size]
+                new_weight = torch.empty((
+                    weight_shape[0] // mp_size,
+                    weight_shape[1],
+                ),
+                                         device=child.weight.device,
+                                         dtype=child.weight.dtype)
+                (query, key, value) = child.weight.chunk(3, dim=0)
+                dst_shape = weight_shape[0] // mp_size // 3
+                qkv = torch.concat((query[mp_replace.gpu_index * dst_shape:(mp_replace.gpu_index + 1) * dst_shape, :],
+                                    key[mp_replace.gpu_index * dst_shape:(mp_replace.gpu_index + 1) * dst_shape, :],
+                                    value[mp_replace.gpu_index * dst_shape:(mp_replace.gpu_index + 1) * dst_shape, :]),
+                                   dim=0)
```

Comment:
may I ask that you use the `strided_copy` from `mp_replace` class (or add a new function) just to simplify the logic here?